File: payroll-api/src/main/java/com/company/payroll/service/StaffBankingInfoService.java
package com.company.payroll.service;

import java.util.Optional;

import com.company.payroll.model.StaffBankingInfo;
import com.github.pagehelper.PageInfo;

public interface StaffBankingInfoService {

	/**
	 * Method of retrieve staff banking info object based of banking info id.
	 * @param bId banking info id
	 * @return {@link StaffBankingInfo} object
	 */
	Optional<StaffBankingInfo> findById(Integer bId);
	
	/**
	 * Method of listing staff banking info
	 * @param page page number
	 * @param offset page data limit
	 * @return List of StaffBankingInfo
	 */
	PageInfo<StaffBankingInfo> list(int page, int offset);

	/**
	 * Method of updating staff banking info based on banking info id
	 * @param staffBankingInfo {@link StaffBankingInfo} object
	 * @return StaffBankingInfo object after updated.
	 */
	StaffBankingInfo update(StaffBankingInfo staffBankingInfo);

}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/service/StaffBankingInfoService.kt
package com.company.payroll.service

import com.company.payroll.model.StaffBankingInfo

import com.github.pagehelper.PageInfo

interface StaffBankingInfoService {
  /**
   *
   * @param bid
   * @return
   */
  fun findById(bid: Int): StaffBankingInfo

  /**
   * @param page
   * @param offset
   * @return
   */
  fun list(page: Int, offset: Int): PageInfo<StaffBankingInfo>

  /**
   *
   * @param staffBankingInfo
   * @return
   */
  fun update(staffBankingInfo: StaffBankingInfo): StaffBankingInfo
}